packages/vite/src/node/server/ws.ts

```diff
@@ -30,10 +35,32 @@ export function createWebSocketServer(
       }
     })
   } else {
+    const websocketServerOptions: WebSocket.ServerOptions = {}
+    const port = hmr ? hmr.port : 24678
+    if (httpsOptions) {
+      // if we're serving the middlewares over https, the ws library doesn't support automatically creating an https server, so we need to do it ourselves
+      // create an inline https server and mount the websocket server to it
+      const httpsServer = createHttpsServer(httpsOptions, (req, res) => {
+        const UPGRADE_REQUIRED = 426
+
+        const body = STATUS_CODES[UPGRADE_REQUIRED]!
+
+        res.writeHead(UPGRADE_REQUIRED, {
+          'Content-Length': body!.length,
```

Comment:
**suggestion/question:** Can we try to omit non-null assertions?

Maybe just check that body is defined and otherwise throw an error or log a warning or anything else?

Reply:
Alternatively we could just do:

```ts
const code = 426
const body = 'Upgrade Required'
```

that's what's in the `STATUS_CODES` enum anyway.